Error updating dirty state editor

After updating to Elysium 0.5.1 and Xtext 2.8.3, when editing [my latest composition](https://github.com/thSoft/Free-Sheet-Music/tree/master/Harmath%20De%CC%81nes/112.%20zsolta%CC%81r), I get a modal error dialog saying `Error updating dirty state editor` from `org.eclipse.xtext.ui.` on every build, which is kind of showstopper. I'm not sure whether this is a bug in Elysium or Xtext, this didn't occur with Elysium 0.5.0 and Xtext 2.7.2. Stack trace:
```
org.eclipse.emf.common.util.WrappedException: java.lang.IllegalArgumentException: Illegal character in path at index 1: 1 Beatus vir.ily
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:233)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.doResolveLazyCrossReference(LazyLinkingResource.java:192)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReference(LazyLinkingResource.java:151)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.resolveLazyCrossReferences(LazyLinkingResource.java:137)
	at org.eclipse.xtext.EcoreUtil2.resolveLazyCrossReferences(EcoreUtil2.java:528)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription.computeReferenceDescriptions(DefaultResourceDescription.java:135)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription$2.get(DefaultResourceDescription.java:123)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription$2.get(DefaultResourceDescription.java:1)
	at org.eclipse.xtext.util.OnChangeEvictingCache.get(OnChangeEvictingCache.java:77)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescription.getReferenceDescriptions(DefaultResourceDescription.java:120)
	at org.eclipse.xtext.resource.DescriptionUtils.collectOutgoingReferences(DescriptionUtils.java:29)
	at org.eclipse.xtext.resource.impl.DefaultResourceDescriptionManager.isAffected(DefaultResourceDescriptionManager.java:117)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport.isReparseRequired(DirtyStateEditorSupport.java:625)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob$1.exec(DirtyStateEditorSupport.java:157)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob$1.exec(DirtyStateEditorSupport.java:1)
	at org.eclipse.xtext.resource.OutdatedStateManager.exec(OutdatedStateManager.java:121)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.internalReadOnly(XtextDocument.java:520)
	at org.eclipse.xtext.ui.editor.model.XtextDocument$XtextDocumentLocker.readOnly(XtextDocument.java:492)
	at org.eclipse.xtext.ui.editor.model.XtextDocument.readOnly(XtextDocument.java:133)
	at org.eclipse.xtext.ui.editor.DirtyStateEditorSupport$UpdateEditorStateJob.run(DirtyStateEditorSupport.java:142)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
Caused by: java.lang.IllegalArgumentException: Illegal character in path at index 1: 1 Beatus vir.ily
	at java.net.URI.create(URI.java:852)
	at java.net.URI.resolve(URI.java:1036)
	at org.elysium.importuri.LilyPondImportUriResolver.resolve(LilyPondImportUriResolver.java:49)
	at org.eclipse.xtext.scoping.impl.ImportUriResolver.apply(ImportUriResolver.java:56)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider.getAllImportedResources(LilyPondImportUriGlobalScopeProvider.java:136)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider.getAllImportedResources(LilyPondImportUriGlobalScopeProvider.java:142)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider$1.get(LilyPondImportUriGlobalScopeProvider.java:98)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider$1.get(LilyPondImportUriGlobalScopeProvider.java:1)
	at org.eclipse.xtext.util.OnChangeEvictingCache.get(OnChangeEvictingCache.java:77)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider.getImportedUris(LilyPondImportUriGlobalScopeProvider.java:94)
	at org.elysium.scoping.LilyPondImportUriGlobalScopeProvider.getScope(LilyPondImportUriGlobalScopeProvider.java:77)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeProvider.getScope(AbstractGlobalScopeProvider.java:49)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeDelegatingScopeProvider.getGlobalScope(AbstractGlobalScopeDelegatingScopeProvider.java:46)
	at org.eclipse.xtext.scoping.impl.AbstractGlobalScopeDelegatingScopeProvider.getGlobalScope(AbstractGlobalScopeDelegatingScopeProvider.java:42)
	at org.eclipse.xtext.scoping.impl.SimpleLocalScopeProvider.getScope(SimpleLocalScopeProvider.java:62)
	at org.eclipse.xtext.scoping.impl.AbstractDeclarativeScopeProvider.delegateGetScope(AbstractDeclarativeScopeProvider.java:72)
	at org.eclipse.xtext.scoping.impl.AbstractDeclarativeScopeProvider.getScope(AbstractDeclarativeScopeProvider.java:104)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getScope(DefaultLinkingService.java:59)
	at org.eclipse.xtext.linking.impl.DefaultLinkingService.getLinkedObjects(DefaultLinkingService.java:119)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:250)
	at org.eclipse.xtext.linking.lazy.LazyLinkingResource.getEObject(LazyLinkingResource.java:225)
	... 20 more
Caused by: java.net.URISyntaxException: Illegal character in path at index 1: 1 Beatus vir.ily
	at java.net.URI$Parser.fail(URI.java:2848)
	at java.net.URI$Parser.checkChars(URI.java:3021)
	at java.net.URI$Parser.parseHierarchical(URI.java:3105)
	at java.net.URI$Parser.parse(URI.java:3063)
	at java.net.URI.<init>(URI.java:588)
	at java.net.URI.create(URI.java:850)
	... 40 more
```

--- a/org.elysium.test/src/org/elysium/test/regression/GrammarRegressions.xtend
+++ b/org.elysium.test/src/org/elysium/test/regression/GrammarRegressions.xtend
@@ -359,7 +359,7 @@ class GrammarRegressions {
 
 	@Test
 	def void assignmentIncluded() {
-		val filename = "foo.ly";
+		val filename = "1 foo.ly";
 		Files.writeStringIntoFile(filename, ASSIGNMENT);
 		'''
 			\include "«filename»"

--- a/org.elysium.ui/src/org/elysium/ui/hyperlinks/LilyPondHyperlinkHelper.java
+++ b/org.elysium.ui/src/org/elysium/ui/hyperlinks/LilyPondHyperlinkHelper.java
@@ -71,7 +71,8 @@ public class LilyPondHyperlinkHelper extends HyperlinkHelper {
 					if (includedResource != null) {
 						uriToOpen = includedEResource.getURI();
 					}
-				}else{
+				}
+				if(uriToOpen==null){
 					uriToOpen=URI.createURI(uriResolver.resolve(include));
 					if(!uriToOpen.isFile() || uriToOpen.isRelative()){
 						uriToOpen=null;

--- a/org.elysium/src/org/elysium/importuri/LilyPondImportUriResolver.java
+++ b/org.elysium/src/org/elysium/importuri/LilyPondImportUriResolver.java
@@ -7,7 +7,6 @@ import java.io.FileFilter;
 import java.net.URI;
 import java.util.List;
 
-
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IWorkspaceRoot;
 import org.eclipse.core.resources.ResourcesPlugin;
@@ -16,10 +15,10 @@ import org.eclipse.emf.ecore.EObject;
 import org.eclipse.xtext.scoping.impl.ImportUriResolver;
 
 import com.google.common.base.Function;
+import com.google.common.base.Optional;
 import com.google.common.cache.CacheBuilder;
 import com.google.common.cache.CacheLoader;
 import com.google.common.cache.LoadingCache;
-import com.google.common.base.Optional;
 import com.google.common.collect.Lists;
 import com.google.inject.Inject;
 
@@ -27,6 +26,8 @@ import com.google.inject.Inject;
  * Resolves importURIs by first searching in LilyPond's default include path.
  */
 public class LilyPondImportUriResolver extends ImportUriResolver {
+
+	private static final boolean IS_WINDOWS = Optional.fromNullable(System.getProperty("os.name")).or("another").toLowerCase().contains("win");//$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$
 
 	@Inject
 	private ILilyPondPathProvider lilyPondPathProvider;
@@ -46,7 +47,7 @@ public class LilyPondImportUriResolver extends ImportUriResolver {
 			searchUris.add(defaultSearchUri);
 
 			for (URI searchUri : searchUris) {
-				URI resolvedImportUri = searchUri.resolve(importUri);
+				URI resolvedImportUri = saferResolve(searchUri, importUri);
 				File importedFile = new File(resolvedImportUri);
 				if (importedFile.exists()) {
 					return resolvedImportUri.toString();
@@ -57,17 +58,27 @@ public class LilyPondImportUriResolver extends ImportUriResolver {
 		return importUri;
 	}
 
+	private URI saferResolve(URI base, String importUri){
+		URI resolvedImportUri = base.resolve(org.eclipse.emf.common.util.URI.encodeOpaquePart(importUri, true));
+		boolean needToHandleAbsoluteWindowsLocation=IS_WINDOWS && resolvedImportUri.getScheme()!=null && !"file".equals(resolvedImportUri.getScheme());//$NON-NLS-1$
+		if(needToHandleAbsoluteWindowsLocation){
+			return URI.create("file:/"+resolvedImportUri.toString());//$NON-NLS-1$
+		}else{
+			return resolvedImportUri;
+		}
+	}
+
 	private Optional<String> getFileUriOutsideWorkspace(EObject context, String importUri){
-		if(importUri.contains("..")){
+		if(importUri.contains("..")){//$NON-NLS-1$
 			org.eclipse.emf.common.util.URI currentResourceUri = context.eResource().getURI();
 			if(currentResourceUri!=null && currentResourceUri.isPlatform()){
 				File currentResourceAsFile = new File(Platform.getLocation()+currentResourceUri.toPlatformString(false));
 				if (currentResourceAsFile.exists()) {
-					URI resolvedImportUri = currentResourceAsFile.toURI().resolve(importUri);
+					URI resolvedImportUri = saferResolve(currentResourceAsFile.toURI(), importUri);
 					IWorkspaceRoot workspaceRoot= ResourcesPlugin.getWorkspace().getRoot();
 					IFile[] importedFileFoundInWorkspace = workspaceRoot.findFilesForLocationURI(resolvedImportUri);
 					if(importedFileFoundInWorkspace.length==0){
-						File importedFile = new File(currentResourceAsFile.toURI().resolve(importUri));
+						File importedFile = new File(resolvedImportUri);
 						if (importedFile.exists()) {
 							return Optional.of(resolvedImportUri.toString());
 						}
@@ -86,8 +97,8 @@ public class LilyPondImportUriResolver extends ImportUriResolver {
 		@Override
 		public URI load(String key) throws Exception {
 			URI lilypondUri = stringToUri(key);
-			URI shareBaseUri=lilypondUri.resolve("../share/lilypond/");
-			final String lyDirectoryName="ly";
+			URI shareBaseUri=lilypondUri.resolve("../share/lilypond/");//$NON-NLS-1$
+			final String lyDirectoryName="ly";//$NON-NLS-1$
 			File f=new File(shareBaseUri);
 			if (f.isDirectory()){
 				for (File subDir : f.listFiles()) {

